frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/PipelineBuilderForm.tsx

```diff
@@ -120,15 +122,24 @@ const PipelineBuilderForm: React.FC<PipelineBuilderFormProps> = (props) => {
     <YAMLEditorField name="yamlData" model={PipelineModel} onSave={handleSubmit} />
   );
 
-  const sanitizeToForm = (yamlPipeline: PipelineKind) => {
-    const { finally: finallyTasks, ...pipelineSpecProperties } = yamlPipeline.spec;
-
-    const newFormData = {
+  const sanitizeToForm = async (yamlPipeline: PipelineKind) => {
+    const { finally: finallyTasks, ...specProps } = yamlPipeline.spec;
+    const pipelineSpecProperties = _.omit(specProps, ['listTasks', 'finallyListTasks']);
+    const newFormData: PipelineBuilderFormValues = {
       ...formData,
       ...pipelineSpecProperties, // support & keep unknown values as well as whatever they may have changed that we use
       name: yamlPipeline.metadata?.name,
       finallyTasks,
     };
+
+    try {
+      await pipelineBuilderFormSchemaLite().validate(yamlPipeline, {
```

Comment:
Interesting... didn't think of validating the yaml... maybe we change the name of the schema to meet that.

a la `pipelineBuilderYAMLSchema` or something.